WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -435,4 +454,108 @@ - (void)defaultAccountDidChange:(NSNotification *)notification
+    if (count == 0) {
+        self.notificationBadgeIconView.hidden = YES;
+        return;
+    }
+
+    BOOL wasNotificationBadgeHidden = self.notificationBadgeIconView.hidden;
+    self.notificationBadgeIconView.hidden = NO;
+    if (wasNotificationBadgeHidden) {
+        [self animateNotificationBadgeIcon];
+    }
+}
+
+#pragma mark - NSObject(NSKeyValueObserving) Helpers
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
+{
+    if ([keyPath isEqualToString:NSStringFromSelector(@selector(applicationIconBadgeNumber))]) {
+        [self updateNotificationBadgeVisibility];
+    }
+}
+
+#pragma mark - Notification Badge Icon Management
+
+- (void)addNotificationBadgeIcon
+{
+    self.notificationBadgeIconView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, kNotificationBadgeIconSize, kNotificationBadgeIconSize)];
+    self.notificationBadgeIconView.image = [UIImage imageWithColor:[WPStyleGuide jazzyOrange]
+                                                        havingSize:CGSizeMake(kNotificationBadgeIconSize, kNotificationBadgeIconSize)];
+    self.notificationBadgeIconView.layer.cornerRadius = kNotificationBadgeIconSize / 2.0;
+    self.notificationBadgeIconView.layer.masksToBounds = YES;
+    self.notificationBadgeIconView.layer.borderColor = [[UIColor whiteColor] CGColor];
+    self.notificationBadgeIconView.layer.borderWidth = 1.0;
+    self.notificationBadgeIconView.hidden = YES;
+    [self.view addSubview:self.notificationBadgeIconView];
+
+    [self updateNotificationBadgeIconPosition];
+    [self updateNotificationBadgeVisibility];
+}
+
+- (void)updateNotificationBadgeIconPosition
+{
+    UIInterfaceOrientation orientation = [[UIApplication sharedApplication] statusBarOrientation];
+    BOOL isLandscape = (orientation == UIInterfaceOrientationLandscapeLeft) || (orientation == UIInterfaceOrientationLandscapeRight);
+
+    // We need to take the extra space before & after the tabbar into account for iPad and iPhone 6 Plus
+    CGFloat horizontalOffset = 0.0;
+    if (IS_IPAD) {
+        horizontalOffset = isLandscape ? kNotificationBadgeIconHorizontalOffsetForIPadInLandscape : kNotificationBadgeIconHorizontalOffsetForIPadInPortrait;
+    }
+    else if (isLandscape && WPDeviceIdentification.isiPhoneSixPlus) {
+        horizontalOffset = kNotificationBadgeIconHorizontalOffsetForIPhone6PlusInLandscape;
+    }
+    CGFloat verticalPosition = CGRectGetHeight(self.view.bounds) - (self.tabBar.frame.size.height - kNotificationBadgeIconVerticalOffsetFromTop);
+    // Subtract the space before & after the tabbar
+    CGFloat tabBarContentWidth = self.tabBar.frame.size.width - (horizontalOffset * 2);
+    CGFloat tabItemWidth = tabBarContentWidth / self.tabBar.items.count;
+    CGFloat horizontalPosition = ((WPTabNotifications + 0.5) * tabItemWidth) - kNotificationBadgeIconHorizontalOffsetFromCenter + horizontalOffset;
```

Comment:
Unfortunately, both the tab width and the space before & after the tab content is different in different devices and orientations. That's why I added so many screenshots of different devices and both orientations. I think with this type of solution, this is the best we're going to get. To be perfectly honest, I hate this fact as much as you do @jleandroperez, if not more! So, if you do have any alternative solution suggestions, I am all ears!